drivers/kodi/device.js

```diff
@@ -28,80 +28,113 @@ class KodiDevice extends Homey.Device {
-                    // Check if we are already polling
-                    if(!device.reconnectTimer) {
-                        device.log('Connection lost to ', ipAddress)
-                        device.error(error)
-                        device.setUnavailable()
-                        device.reconnectTimer = 
-                            setTimeout( (ipAddress, port) => {
-                                device._connectKodi(ipAddress, port)
-                            }, RECONNECT_INTERVAL, ipAddress, port) // Start polling
-                    }                    
-                }
-
-                // Register event listeners
-                kodi.on('close', fnReconnect)
-                kodi.on('error', fnReconnect)
-
-                // Register objects and events to interact with kodi
-                device._player = new Player(kodi)
-                device._player.on('pause', () => { device._onPause() })
-                device._player.on('stop', () => { device._onStop() })
-                device._player.on('episode_stop', (episode) => { device._onEpisodeStop(episode) })
-                device._player.on('movie_stop', (movie) => { device._onMovieStop(movie) })
-                device._player.on('play', () => { device._onPlay() })
-                device._player.on('resume', () => { device._onResume() })
-                device._player.on('movie_start', (movie) => { device._onMovieStart(movie) })
-                device._player.on('episode_start', (episode) => { device._onEpisodeStart(episode) })
-                device._player.on('song_start', (song) => { device._onSongStart(song) })
-
-                device._library = new Library(kodi)
-
-                device._system = new System(kodi)
-                device._system.on('shutdown', () => { device._onShutdown() })
-                device._system.on('hibernate', () => { device._onHibernate() })
-                device._system.on('reboot', () => { device._onReboot() })
-                device._system.on('wake', () => { device._onWake() })
-
-                device.setAvailable() // Make available to Homey
-            })
-            .catch((err) => {
-                device.error(err)
-                device.setUnavailable(err)
-                device.reconnectTimer = 
-                    setTimeout( (ipAddress, port) => {
-                        device._connectKodi(ipAddress, port)
-                    }, RECONNECT_INTERVAL, ipAddress, port) // Start polling                
-            })
+    handleDisconnect({ ipAddress, port, err } = {}) {
+        this.log('got disconnected');
+        if (this._kodi) {
+            this.log('...cleaning up');
+            this.cleanup();
+        }
+        if (err) {
+            this.error(err)
+        }
+        this.setUnavailable(err || '')
+        this.reconnectTimer && clearTimeout(this.reconnectTimer);
+        this.reconnectTimer = setTimeout(() => {
```

Comment:
setInterval()

When the kodi device gets disconnected, it needs to **periodically** poll, not just once. As far as I know setTimeout() only fires once